neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -183,33 +213,25 @@ def _parse_header(self):
         event_annotations = []
 
         if self.rawmode == "one-dir":
-            filenames = sorted(os.listdir(self.dirname))
-            dirname = self.dirname
+            filenames = os.listdir(self.dirname)
         else:
-            if not os.path.isfile(self.filename):
+            filenames = self.include_filenames
+
+        filenames = [f for f in filenames if f not in self.exclude_filenames]
+        full_filenames = [os.path.join(self.dirname, f) for f in filenames]
+
+        for filename in full_filenames:
+            if not os.path.isfile(filename):
                 raise ValueError(
                     f"Provided Filename is not a file: "
-                    f"{self.filename}. If you want to provide a "
+                    f"{filename}. If you want to provide a "
                     f"directory use the `dirname` keyword"
                 )
 
-            dirname, fname = os.path.split(self.filename)
-            filenames = [fname]
-
-        if not isinstance(self.exclude_filename, (list, set, np.ndarray)):
-            self.exclude_filename = [self.exclude_filename]
-
-        # remove files that were explicitly excluded
-        if self.exclude_filename is not None:
-            for excl_file in self.exclude_filename:
-                if excl_file in filenames:
-                    filenames.remove(excl_file)
-
         stream_props = {}  # {(sampling_rate, n_samples, t_start): {stream_id: [filenames]}
 
-        for filename in filenames:
-            filename = os.path.join(dirname, filename)
-
+        for filename in full_filenames:
+            filename = os.path.join(self.dirname, filename)
```

Comment:
```suggestion
        for filename in full_filenames:
```

Since you did this in L221 you shouldn't need to do this again right?

Reply:
Yes indeed! Thanks for pointing this out!